### PR TITLE
board/native: adapted Makefile.dep for netif init

### PR DESCRIPTION
**Pull request by haukepetersen**
The new network stack does not now the concept of the defaulttransceiver, so this fix makes auto_init_ng_netif work for now.

### CODE REVIEW

**Comment by authmillenon** [comment]
Is this only needed for native, or are the other boards elided for a different reason?

**Comment by haukepetersen** [comment]
Actually, I have not looked at other boards (as the `auto_init_ng_netif` is working with all other boards)... So I guess native is the only board right now, that needs this fix.

**Comment by haukepetersen** [comment]
I updated it again slightly (and accidentally squashed already), now the Makefile has the identical structure of the `Makefile.dep` of the `iot-lab` and the `samr21`.

**Comment by authmillenon** [comment]
Can't make line comments on mobile it seems so I have to comment here: Maybe out of scope of this PR, but I'm still opposed to using `ng_netif`as the signifier to include `ng_nativenet` (or some radio on other boards) et al. What if I just want to test netif using unittests? What if I implement some shiny new netapi-driven interface on native, like IP over raw sockets? What if I don't want to use nomac as the MAC layer? I prefer to keep the name `defaulttransceiver` or introduce a new `defaultif` pseudo module as a signifier for this behavior.

**Comment by haukepetersen** [comment]
I actually agree, but I would like to continue this discussion somewhere else and merge this PR ASAP which then enables the merging of the `ng_networking` example which I want to get done quickly. Does this work for you?

**Comment by authmillenon** [comment]
Yes, ACK

**Comment by haukepetersen** [comment]
ups, there was a still a bug in `Makefile.dep` which lead to some failing builds for Travis. Its fixed now and I added a slight improvement for the error message that showed the bug. 
@authmillenon, do you re-ack?

**Comment by authmillenon** [comment]
Re-ACK

**Comment by authmillenon** [comment]
Now we have we see the fruits of using `ng_netif` as the signifiers: unittests need the tap interface as an argument now -.-

**Comment by haukepetersen** [comment]
To circumvent the problem above, I introduced a new pseudo-module called `ng_netif_default`. This is basically the same as the `defaulttransceiver`. I just did not like the old name, as this module can target more than 1 transceiver and it is also tied to the gnrc network stack, so I think something gnrc specific is a must.

File: boards/native/Makefile.dep [hunk 1]
```diff
@@ -1,9 +1,13 @@
 ifneq (,$(filter defaulttransceiver,$(USEMODULE)))
-    ifeq (,$(filter ng_netif,$(USEMODULE)))
-        USEMODULE += nativenet
+    USEMODULE += nativenet
+    ifeq (,$(filter netdev_base,$(USEMODULE)))
```
Comment (authmillenon) [inline boards/native/Makefile.dep]: what's `netdev_base`

Comment (authmillenon) [inline boards/native/Makefile.dep]: Ah, got it… a relic from when `netdev` was super complicated :D

Comment (haukepetersen) [inline boards/native/Makefile.dep]: you should know this... I just copied it from the Makefile.deps of the iot-lab and the samr21...

**Comment by authmillenon** [comment]
ACK

**Comment by authmillenon** [comment]
&go

**Pull request merged**